dags/public-health/Code55.py

```diff
@@ -0,0 +1,152 @@
+from datetime import datetime, timedelta
+
+import pandas as pd
+import requests
+
+from airflow import DAG
+from airflow.hooks.postgres_hook import PostgresHook
+from airflow.operators.python_operator import PythonOperator
+
+default_args = {
+    "owner": "airflow",
+    "depends_on_past": False,
+    "start_date": datetime(2020, 1, 30),
+    "email": [
+        "ian.rose@lacity.org",
+        "hunter.owens@lacity.org",
+        "brendan.bailey@lacity.org",
+    ],
+    "email_on_failure": True,
+    "email_on_retry": False,
+    "retries": 1,
+    "retry_delay": timedelta(days=1),
+}
+
+dag = DAG("code-55", default_args=default_args, schedule_interval="@daily")
+
+
+# Code to retrieve/clean data from each of the API endpoints
+def get_code_55_df(urlnumber, stage):
+    # make request to feature service url
+    url = (
+        r"https://production.sanstarla.com/FeatureService/REST/services/Authorizations/FeatureServer/%s/query" % urlnumber
+    )
+    resp = requests.get(url)
+
+    # store raw data into JSON format
+    # isolate desired values from 'features' dict.keys()
+    raw_data = resp.json()
+    data = raw_data.get("features")
+    # create new empty list to store desired final data
+    cleandict = []
+    # iterate through 'features' dict.items() which are nested dictionaries
+    # isolate 'attributes' dict.items() and append to empty list
+    for d in data:
+        cleandict.append(d.get("attributes"))
+
+    # load list of dictionaries into a pandas dataframe and convert to CSV file
+    df = pd.DataFrame(cleandict)
+
+    # Deduplicating DataFrame
+    df["CompositeID"] = (
+        df["Number"].astype(str) + "-" + df["AssociatedServiceRequestNumber"]
+    )
+    df.drop_duplicates(subset="CompositeID", inplace=True)
+
+    # Assigning dataset
+    df["Stage"] = stage
+
+    # Adjusting datetimes from unix timestamp to readable date
+    datelist = [
+        "DateCreated",
+        "DateApproved",
+        "ExpirationDate",
+        "DateCompleted",
+        "CleanupDate",
+    ]
+    for date in datelist:
+        df[date] = pd.to_datetime(df[date], unit="ms")
+    return df
+
+
+def update_code_55():
```

Comment:
I think this function needs to take `**kwargs`, since it is passed a bundle of them by the scheduler.

Reply:
Adding now, but the python operator doesn't use `op_kwargs`. What additional arguments would the scheduler be passing?

`t1 = PythonOperator(
    task_id="Update_Code_55_Data",
    provide_context=True,
    python_callable=update_code_55,
    dag=dag,
)`

Reply:
I'm not sure the full list of kwargs, but this errored when I was testing because it was at least passing the `dag`.